projects/client/RabbitMQ.Client/src/client/impl/TcpClientAdapterHelper.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+
+namespace RabbitMQ.Client
+{
+    public static class TcpClientAdapterHelper
+    {
+        public static IPAddress GetMatchingHost(IReadOnlyCollection<IPAddress> addresses, AddressFamily addressFamily)
+        {
+            var ep = addresses.FirstOrDefault(a => a.AddressFamily == addressFamily);
+            if (ep == null && addresses.Count == 1 && addressFamily == AddressFamily.Unspecified)
```

Comment:
Let's make this `if (ep == null) return addresses.FirstOrDefault()` - I'm not convinced we need all the other checks - might as well try to return something if we got it.

Reply:
@kjnilsson I agree that check for `address.Count` may be too restrictive, but I think we only should perform it if socket is `Unspecified` because otherwise we may return `ipv4` address when socket explicitely says it's `ipv6` or something

Reply:
ok makes sense